engine/ipc.go

```diff
@@ -59,7 +78,7 @@ func (s *SeesawEngine) Failover(ctx *ipc.Context, reply *int) error {
 		return errContext
 	}
 
-	if !ctx.IsTrusted() {
+	if !(ctx.IsAuthenticated() && ctx.User.IsAdmin() || ctx.IsTrusted()) {
```

Comment:
Super quick glance, but these seem like they should be moved to functions on the ctx to avoid repeating the `!(ctx.IsAuthenticated() && ctx.User.IsSomething() || ctx.IsTrusted' dance - `ctx.CanRead`, `ctx.CanWrite` or similar may work for naming.

Reply:
I like it. Done.